### ISSUE TITLE
Cannot rename folder

### ISSUE THREAD

**Issue by taisho2023** [SpyglassMC/Spyglass#1414]
> [!NOTE]
> This issue is caused on Windows due to the file watcher we are using. **You can try the following workaround**:
> 1. Create a file `spyglass.json` in the workspace root
> 2. Paste the following contents inside:
> ```json
> {
>   "env": {
>     "useFilePolling": true
>   }
> }
> ```
> 3. Reload vscode, for example by running the `Developer: Reload Window` command after pressing Ctrl+Shift+P

https://github.com/SpyglassMC/Spyglass/assets/154045603/95f0fbd5-9ede-4684-9088-90f21432d903

**Comment by misode** [comment]
This seems like a problem with chokidar, as explained here: https://github.com/paulmillr/chokidar/issues/664#issuecomment-356144077

Unless we switch to an entirely different way of watching files I don't think this is likely to be solved

**Comment by misode** [comment]
@SPGoding What were the problems with vscode's `createFileSystemWatcher` API that made you switch to chokidar?

**Comment by SPGoding** [comment]
There is a [`DidChangeWatchedFiles` notification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeWatchedFiles) in LSP so I believe editor support is not as big of a problem (depending on how many editors implement that notification). I think I run into issues with how folder operations (like rename and deletes) are notified in the events. I believe the chokidar watcher we use sends an individual event for every entry in the affected folder recursively while the LSP notification sent by VS Code only includes the top level folder so I had to do some weird heuristics that doesn't really work most (#351).

**Comment by Kesuaheli** [comment]
Oh... that comes from Spyglass...
I really wondered why VSCode wont let me rename any folders. I thought VSCode had a buggy update.
I couldnt even rename them in Windowns Explorer, because of "is opened by another program" (the file watcher, ig?).
Have to close VSCode everytime in order to rename.
Now I found this issue in the Discord gh-issues channel.

+1 to fix this

**Comment by misode** [comment]
Reopening because the config is opt-in and this is a common issue people are experiencing. We should look into a better solution in the future.

**Comment by B1BU** [comment]
I'm having this issue very frequently, can't rename anything with the extension enabled

**Comment by B1BU** [comment]
Is there a way to enable this useFilePolling option globally?

**Comment by misode** [comment]
> Is there a way to enable this useFilePolling option globally?

No, that's not possible. This workaround is not meant to be a final solution, we plan to fix this issue properly in the future.

**Comment by josephNg95** [comment]
I got this issue last month, then I removed VSCode and installed it again, and then it worked.
But now I don't know why it came back, I can not rename a folder :D That's crazy.
